src/coord_seq.rs

```diff
@@ -53,6 +61,80 @@ impl<'a> CoordSeq<'a> {
         }
     }
 
+    /// Creates a new `CoordSeq`.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use geos::CoordSeq;
+    ///
+    /// let coords = CoordSeq::new_from_vec(&[&[0., 1.], &[2., 3.]])
+    ///                       .expect("failed to create CoordSeq");
+    /// assert!(coords.get_y(1) == Ok(3.));
+    ///
+    /// // Doing it from a Vec<Vec<f64>>.
+    /// let positions = vec![vec![0., 1.], vec![2., 3.]];
+    /// let s_positions = positions.iter().map(|x| x.as_slice()).collect::<Vec<_>>();
+    /// let coords = CoordSeq::new_from_vec(&s_positions)
+    ///                       .expect("failed to create CoordSeq");
+    /// assert!(coords.get_y(1) == Ok(3.));
+    ///
+    /// // All vectors don't have the same length, this is an error!
+    /// assert!(CoordSeq::new_from_vec(&[&[0., 1.], &[3.]]).is_err());
+    ///
+    /// // An empty vector is an error as well since we can't figure out its dimensions!
+    /// assert!(CoordSeq::new_from_vec(&[]).is_err());
+    /// ```
+    pub fn new_from_vec(data: &[&[f64]]) -> GResult<CoordSeq<'a>> {
+        let size = data.len();
+
+        if size > 0 {
+            let dims = data[0].len();
+            if let Err(e) = CoordDimensions::try_from(dims as _) {
+                return Err(Error::GenericError(e.to_owned()));
+            }
+            if !data.iter().skip(1).all(|x| x.len() == dims) {
+                return Err(Error::GenericError("All vec entries must have the same size!".to_owned()));
+            }
+            match match GContextHandle::init() {
```

Comment:
wouldn't it be more straighforward to write this with a `and_then` ?

I find the `match match` a bit difficult to read

Reply:
Good suggestion! I just like chaining `match` but you're right!